README.rst

```diff
@@ -68,51 +62,162 @@ the type of context being adapted directly. For example::
-exists for the context - you can adapt a context to ILockingSupport as
-normal::
+After this is done you can adapt a context to ``ILockingSupport`` as normal::
 
     locking = ILockingSupport(context, None)
 
     if locking is not None:
         locking.lock()
 
-You'll get an instance of LockingSupport if context can be adapted to
-IBehaviorAssignable (which, recall, is application specific), and if the
-implementation of IBehaviorAssignable says that this context supports this
-particular behavior.
+You'll get an instance of ``LockingSupport`` if context can be adapted to ``IBehaviorAssignable`` (which, recall, is application specific),
+and if the implementation of ``IBehaviorAssignable`` says that this context supports this particular behavior.
+
+It is also possible to let the provided interface act as a marker interface that is to be provided directly by the instance.
+To achieve this, omit the ``factory`` argument.
+This is useful if you need to register other adapters (including views and viewlets) for instances providing a particular behavior.
+
+Like the IBehaviorAssignable plumbing, marker interface support needs to be enabled on a per-application basis.
+It can be done with a custom __providedBy__ decorator or an IObjectCreatedEvent handler for applying the marker.
+A sample event handler is provided with this package, but is not registered by default
+
+ZCML Reference
+--------------
+
+The ``plone:behavior`` directive uses the namespace ``xmlns:plone="http://namespaces.plone.org/plone"``.
+In order to enable it loading of its ``meta.zcml`` is needed, use::
+
+    <include package="plone.behavior" file="meta.zcml" />
+
+The directive supports the attributes:
+
+``title``
+    A user friendly title for this behavior (required).
+
+``description``
+    A longer description for this behavior (optional).
+
+``provides``
+    An interface to which the behavior can be adapted.
+    This is what the conditional adapter factory will be registered as providing (required).
+
+``marker``
+    A marker interface to be applied by the behavior.
+    If ``factory`` is not given, then this is optional and defaults to the value of ``provides``.
+    If factory is given ``marker`` is required and should be different from ``provides`` - even if its not enforced.
+
+``factory``
+    The factory for this behavior (optional).
+    If no factory is given, the behavior context is assumed to provide the interface given by ``provides`` itself.
+
+    If factory provides ``plone.behavior.interfaces.ISchemaAwareFactory`` the factory is assumed to be a callable.
+    It is called with the value given in ``provides`` as the only argument.
+    The value returned is then used as the factory.
+
+    ``plone.behavior.interfaces.ISchemaAwareFactory`` is a marker interface for factories that should be initialised with a schema.
+    Its expected to return another callable that can create appropriate behavior factories on demand.
```

Comment:
It**s**, or It is. But it seems this paragraph is largely repeating the previous one. Just remove it?